provider/mesh/peer.go

```diff
@@ -301,14 +301,45 @@ func (s *Silences) Mutes(lset model.LabelSet) bool {
 
 // All returns a list of all known silences.
 func (s *Silences) All() ([]*types.Silence, error) {
+	return s.Query(s.Count(), 0)
+}
+
+func (s *Silences) Count() int {
+	return len(s.st.m)
```

Comment:
ah so I guess I didn't think this part through, I assumed it was atomic without checking. I'll add the readlock.